### PR TITLE
CANS-1000: Complete modal window reworked

### PR DESCRIPTION
**Pull request by LeonidMarushevskyi**
## JIRA
https://osi-cwds.atlassian.net/browse/CANS-1000

## Description

Complete warning modal replaced with cwds/components Modal component. Modal text and buttons behaviour changed. Using styles from Component Library.

## Tests

- [ ] I used TDD to develop the feature
- [x] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/Application/components/Assessment/CompleteModal.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { Modal, ModalBody, ModalHeader, CardTitle, Button } from '@cwds/components'
+
+const CompleteModal = ({
+  isCompleteModalShown,
+  handleCompleteWarning,
+  handleCompleteAssessment,
+  handleSaveAssessment,
+}) => (
+  <div>
+    {isCompleteModalShown ? (
```
Comment (denys-davydov) [inline app/javascript/Application/components/Assessment/CompleteModal.js]: We probably do not need the `div` wrapper and `isCompleteModalShown` ternary here.  I believe the `Modal`'s `isOpen` prop will handle it for us. Please try to remove them and use this line for the modal:
`<Modal isOpen={isCompleteModalShown} size="lg">`

Comment (LeonidMarushevskyi) [reply app/javascript/Application/components/Assessment/CompleteModal.js]: Good catch. Done

File: spec/regression/case_worker_scenarios_spec.rb [hunk 2]
```diff
@@ -497,16 +497,18 @@ def click_save_and_summary_card_is_shown
   def warning_and_summary_card_shown_after_complete_button_clicked(is_reassessment)
     expect(@form.header.authorization_radio_no.checked?).to be(true)
     @form.footer.complete_button.click
-    expect(@form.app_globals.complete_warning_modal['style']).to eq('display: block;')
-    @form.app_globals.cancel_button_of_warning.click
+    expect(@form.app_globals).to have_complete_warning_modal
+    sleep(2)
```
Comment (nasar-cambria) [inline spec/regression/case_worker_scenarios_spec.rb]: @zliuwd created a method for sleep and focus. Was that deleted some how I see you are using sleep here?

File: spec/regression/case_worker_scenarios_spec.rb [hunk 3]
```diff
@@ -497,16 +497,18 @@ def click_save_and_summary_card_is_shown
...
+    @form.app_globals.complete_warning_save_return_button.click
     if is_reassessment
       expect(@form.global).to have_reassessment_page_header
     else
       expect(@form.global).to have_assessment_page_header
     end
     expect(@form.footer).to have_complete_button
     @form.footer.complete_button.click
-    @form.app_globals.agree_button_of_warning.click
+    sleep(2)
```
Comment (nasar-cambria) [inline spec/regression/case_worker_scenarios_spec.rb]: Same comment as above

File: spec/regression/case_worker_scenarios_spec.rb (hunk 2 above)
Comment (LeonidMarushevskyi) [reply spec/regression/case_worker_scenarios_spec.rb]: The problem here is not related to scrolling. For some reason, Selenium has an issue with clicking buttons in modal windows from the component library. The temporary solution is explicitly to wait until the modal window will be completely rendered. Will continue looking for a better solution.

**Pull request merged**